src/libraries/System.Private.CoreLib/src/System/Collections/Generic/IDictionaryDebugView.cs

```diff
@@ -17,12 +17,17 @@ public IDictionaryDebugView(IDictionary<K, V> dictionary)
         }
 
         [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
-        public KeyValuePair<K, V>[] Items
+        public DebugViewDictionaryItem<K, V>[] Items
         {
             get
             {
-                KeyValuePair<K, V>[] items = new KeyValuePair<K, V>[_dict.Count];
-                _dict.CopyTo(items, 0);
+                var keyValuePairs = new KeyValuePair<K, V>[_dict.Count];
```

Comment:
Is the intermediate KeyValuePair array necessary here? Seems it could be avoided.

Reply:
The only way I found to avoid creating this temporary array was to enumerate the dictionary. However, such a change slowed down the code by at least 30%. I think this would affect the debugger UI responsiveness more than the array allocation. Have you thought about some other solution?

Reply:
Right, I incorrectly assumed this was only targeting `Dictionary<K,V>` and missed it's targeting the interface. Looks good then.